subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectContainerExtensions.kt

```diff
@@ -41,6 +43,26 @@ inline operator fun <T : Any, C : NamedDomainObjectContainer<T>> C.invoke(
 }
 
 
+/**
+ * Allows the container to be configured via an augmented DSL.
+ *
+ * @param configuration The expression to configure this container with
+ * @return The container.
+ */
+@Suppress("nothing_to_inline")
+@Deprecated(level = DeprecationLevel.HIDDEN, message = "For backward compatibility only")
+inline operator fun <T : Any, C : NamedDomainObjectContainer<T>> C.invoke(
+    configuration: NamedDomainObjectContainerScope<T>.() -> Unit
+): C = apply {
+    DeprecationLogger.deprecateInvocation("NamedDomainObjectContainer<T>.invoke(kotlinLambda)")
+        .withAdvice("Recompile your plugin against Gradle >= 6.8")
+        .willBeRemovedInGradle8()
+        .undocumented()
```

Comment:
Can we add a short note in the upgrade guide and link to that here instead of adding an undocumented deprecation?

Reply:
It feels like undue cost to me.

Reply:
I take it back. This will start showing up for protobuf plugin users so it makes sense to have an explanation somewhere.

Reply:
Done. PTAL, @big-guy.